### ISSUE TITLE
Deriving `TryFromBytes` warns about `non_camel_case_types`

### ISSUE THREAD

**Issue by bluurryy** [google/zerocopy#2051]
When you derive `TryFromBytes` on an enum like this:
```rust
#[repr(u32)]
#[allow(non_camel_case_types)]
#[derive(TryFromBytes)]
pub enum Code {
    I32_ADD,
    I32_SUB,
    I32_MUL,
}
```
You get warnings about `non_camel_case_types` from the expanded code: 
```rust
// ...
#[repr(u32)]
#[allow(dead_code)]
enum ___ZerocopyTag {
    I32_ADD,
    I32_SUB,
    I32_MUL,
}
// ...
```
Adding `#[allow(non_camel_case_types)]` to the `___ZerocopyTag` will fix this.

**Comment by joshlf** [comment]
Thanks for reporting this! I've opened a fix in #2057.

**Comment by joshlf** [comment]
Closed in #2057
